src/content/docs/recipes/handling-dark-mode.mdx

```diff
@@ -0,0 +1,162 @@
+    ```
+
+2. Define a Setter function to store the theme value in local storage
+
+    ```ts title="src/components/theme-toggle/script.js"
+    /**
+    * @param {'dark'|'light'|'auto'} theme
+    */
+    const setTheme = (theme = 'auto') => {
+        localStorage.setItem('theme', theme);
+        theme = theme;
+    }
+    ```
+
+3. if the theme is "dark", add the "dark" class to the `html` element, else remove the "dark" class from `html` element
+
+    ```ts title="src/components/theme-toggle/script.js"
+    /**
+    * @param {'dark'|'light'|'auto'} theme
+    */
+    const setTheme = (theme = 'auto') => {
+        localStorage.setItem('theme', theme);
+        theme = theme;
+        if (theme === 'dark') 
+            document.documentElement.classList.add('dark');
+        else if (theme === 'light')
+            document.documentElement.classList.remove('dark');
+        else {
+            if(window.matchMedia('(prefers-color-scheme: dark)').matches)
+                document.documentElement.classList.add('dark');
+            else 
+                document.documentElement.classList.remove('dark');
+        }
+    }
+    ```
+
+4. in the same folder define a button with `#theme-toggle` **id** to toggle the theme
+
+    ```html title="src/components/theme-toggle/index.astro"
+    <button id="theme-toggle">Toggle Theme</button>
+    ```
+
+5. Listening for Click Event on the button with `#theme-toggle` **id** to toggle the theme
+    <Tabs>
+        <TabItem label="MPA">
+            ```ts title="src/components/theme-toggle/script.js"
+            const options = ['dark', 'light', 'auto'];
+            setTheme('auto')
+            window.onload = () => {
+                document.querySelector('#theme-toggle')
+                    ?.addEventListener('click', () => {
+                    // rotate through options
+                    setTheme(options[(options.indexOf(theme) + 1) % options.length]);
+                });
+            };
+            ```
+        </TabItem>
+        <TabItem label="VT">
+            ```ts title="src/components/theme-toggle/script.js"
+            const options = ['dark', 'light', 'auto'];
+            setTheme('auto')
+            document.addEventListener('astro:page-load', () => {
+                document.querySelector('#theme-toggle')
+                    ?.addEventListener('click', () => {
+                    // rotate through options
+                    setTheme(options[(options.indexOf(theme) + 1) % options.length]);
+                });
+            });
+            ```
+        </TabItem>
+    </Tabs>
+
+6. Listening for System Changes    
+
+    ```ts
+        window.matchMedia('(prefers-color-scheme: dark)')
+            .addEventListener('change', ({ matches: isDark }) => {
+                setTheme(isDark ? 'dark' : 'light')
+            })
+    ```
+
+7. Load the `toggle-theme/script.js` in the layouts, to make it available in all pages and include `is:inline` attribute to avoid FOUC
```

Comment:
As I said earlier, I'd prefer the first step to specify that the content should be inside an inline script (no src) and get rid of this step